compactstorage.cpp

```diff
@@ -1,182 +1,204 @@
-		numberOfBytes = 1;
-	}
-
-	m_numBytes = numberOfBytes;
-	m_usedBits = 0;
-	m_curPos = 0;
-	allocateBytes(numberOfBytes);
-}
-
-CompactStorage::~CompactStorage()
-{
-	freeBytes();
-}
-
-void CompactStorage::allocateBytes(int numberOfBytes)
-{
-	m_bytes = new char[numberOfBytes];
-
-	for (int i = 0; i < numberOfBytes; i++) {
-		m_bytes[i] = '\0';
-	}
-}
-
-void CompactStorage::freeBytes()
-{
-	delete[] m_bytes;
-}
-
-void CompactStorage::reallocateBytes(int numberOfBytes)
-{
-	char* temp = new char[numberOfBytes];
-	for (int i = 0; i < m_numBytes; i++) {
-		temp[i] = m_bytes[i];
-	}
-	freeBytes();
-	m_bytes = temp;
-	m_numBytes = numberOfBytes;
-}
-
-void CompactStorage::ensureRoomFor(int bits)
-{
-	cout << "make sure we have room for " << bits << " bits..." << endl;
-	int newLength = m_usedBits + bits;
-	if (newLength > m_numBytes * 8) {
-		reallocateBytes(newLength / 8 + 1);
-	}
-}
-
-int CompactStorage::curByte()
-{
-	return m_curPos / 8;
-}
-
-int CompactStorage::curBit()
-{
-	return m_curPos % 8;
-}
-
-void CompactStorage::dump()
-{
-	for (int curByte = 0; curByte < m_numBytes; curByte++) {
-		char str = m_bytes[curByte];
-		for (int i = 128; i > 0; i = i/2) {
-			cout << (bool) ((int) str & i);
-		}
-		cout << " ";
-	}
-	cout << endl;
-	for (int i = 0; i < m_curPos + m_numBytes - 1; i++) {
-		cout << " ";
-	}
-	cout << "^" << endl;
-}
-
-void CompactStorage::writeInt(int value, int bits)
-{
-	ensureRoomFor(bits);
-
-	// Iterate over the data bits that we need to set
-	for (int i = 0; i < bits; i++) {
-		int byte = curByte();
-		int bit = curBit();
-
-		// Create bitmask for the correspondent bit of the integer to store here
-		int mask = 1 << (bits - i - 1);
-
-		// Whether to set the current bit or not
-		bool set = (bool) (value & mask);
-		m_bytes[byte] |= (char) set << (7 - bit);
-		m_curPos++;
-	}
-
-	m_usedBits += bits;
-}
-
-void CompactStorage::writeBool(bool value)
-{
-	char byte = m_bytes[curByte()];
-	byte |= (int) value << (7 - curBit());
-	m_bytes[curByte()] = byte;
-
-	m_curPos += 1;
-	m_usedBits += 1;
-}
-
-int CompactStorage::readInt(int bits)
-{
-	// TODO: Signed vs. unsigned might cause problems here
-	int value = 0;
-
-	// Iterate over the data bits that we need to set
-	for (int i = 0; i < bits; i++) {
-		int byte = curByte();
-		int bit = curBit();
-
-		// Create bitmask for the correspondent bit of the data that we want to read
-		int mask = 1 << (7 - bit);
-
-		// Whether to set the current bit or not
-		bool set = (bool) (m_bytes[byte] & mask);
-		value |= (int) set << (bits - i - 1);
-		m_curPos++;
-	}
-
-	return value;
-}
-
-bool CompactStorage::readBool()
-{
-	unsigned int byte = m_bytes[curByte()];
-	unsigned int mask = 1 << (7 - curBit());
-
-	m_curPos += 1;
-	return (byte & mask) > 0;
-}
-
-char* CompactStorage::getContent(int & length)
-{
-	char* result = new char[m_numBytes];
-	*result = *m_bytes;
-
-	length = m_numBytes;
-
-	return result;
-}
-
-void CompactStorage::reset()
-{
-	m_curPos = 0;
-}
+/*
```

Comment:
It looks like you (accidentally?) changed the line endings of this file. Please change back to CRLF, it makes the diff easier to understand.

Reply:
Yes, it was an accident, I'm sorry. Text editor issues. I'll fix it.